src/test/java/seedu/cakecollate/logic/parser/ParserUtilTest.java

```diff
@@ -50,6 +53,57 @@
+    private static final String VALID_INDEX_LIST_WITH_ONE_INDEX_AND_EXTRA_WHITESPACE_AT_BACK =
+            "1 ";
+    private static final String VALID_INDEX_LIST_WITH_ONE_INDEX_AND_EXTRA_WHITESPACES_AT_FRONT_AND_BACK =
+            " 1 ";
+    private static final String VALID_INDEX_LIST_WITH_TWO_INDEXES_AND_EXTRA_WHITESPACE_AT_FRONT =
+            " 1 2";
+    private static final String VALID_INDEX_LIST_WITH_TWO_INDEXES_AND_EXTRA_WHITESPACE_AT_BACK =
+            "1 2 ";
+    private static final String VALID_INDEX_LIST_WITH_TWO_INDEXES_AND_EXTRA_WHITESPACES_AT_FRONT_AND_BACK =
+            " 1 2 ";
+    private static final String VALID_INDEX_LIST_WITH_THREE_INDEXES_AND_EXTRA_WHITESPACE_AT_FRONT =
+            " 1 2 3";
+    private static final String VALID_INDEX_LIST_WITH_THREE_INDEXES_AND_EXTRA_WHITESPACE_AT_BACK =
+            "1 2 3 ";
+    private static final String VALID_INDEX_LIST_WITH_THREE_INDEXES_AND_EXTRA_WHITESPACES_AT_FRONT_AND_BACK =
+            " 1 2 3 ";
+    private static final String VALID_INDEX_LIST_WITH_ONE_INDEX_AND_MULTIPLE_WHITESPACES_1 =
+            "   1   ";
+    private static final String VALID_INDEX_LIST_WITH_ONE_INDEX_AND_MULTIPLE_WHITESPACES_2 =
+            "            1     ";
+    private static final String VALID_INDEX_LIST_WITH_ONE_INDEX_AND_MULTIPLE_WHITESPACES_3 =
+            "                                1                 ";
+    private static final String VALID_INDEX_LIST_WITH_TWO_INDEXES_AND_MULTIPLE_WHITESPACES_1 =
+            "  1  2  ";
+    private static final String VALID_INDEX_LIST_WITH_TWO_INDEXES_AND_MULTIPLE_WHITESPACES_2 =
+            "  1          2     ";
+    private static final String VALID_INDEX_LIST_WITH_TWO_INDEXES_AND_MULTIPLE_WHITESPACES_3 =
+            "             1                   2                 ";
+    private static final String VALID_INDEX_LIST_WITH_THREE_INDEXES_AND_MULTIPLE_WHITESPACES_1 =
+            "  1  2  3  ";
+    private static final String VALID_INDEX_LIST_WITH_THREE_INDEXES_AND_MULTIPLE_WHITESPACES_2 =
+            "  1          2           3  ";
+    private static final String VALID_INDEX_LIST_WITH_THREE_INDEXES_AND_MULTIPLE_WHITESPACES_3 =
+            "             1                   2          3              ";
+
+    private static final String INVALID_INDEX_LIST_WITH_ONE_INDEX = "-1";
+    private static final String INVALID_INDEX_LIST_WITH_TWO_INDEXES = "-1 -2";
+    private static final String INVALID_INDEX_LIST_WITH_THREE_INDEXES = "-1 -2 -3";
+    private static final String INVALID_INDEX_LIST_WITH_ONE_INDEX_AND_MULTIPLE_SPACES = "  -1  ";
+    private static final String INVALID_INDEX_LIST_WITH_TWO_INDEXES_AND_MULTIPLE_SPACES = "    -1    -2   ";
+    private static final String INVALID_INDEX_LIST_WITH_THREE_INDEXES_AND_MULTIPLE_SPACES = "   -1    -2    -3   ";
+    private static final String INVALID_INDEX_LIST_WITH_SINGLE_WHITESPACE = " ";
+    private static final String INVALID_INDEX_LIST_WITH_MULTIPLE_WHITESPACES = "     ";
```

Comment:
think you can include more invalid test cases and varying the indexes where the error will pop up, and having duplicates 
eg, "-1 2 3"
eg " 1 1 1 1 "
eg "a 1 2"

Reply:
Sure will add these in!